Web_Development/Javascript/4.Programming Fundamentals/5 Type Conversion/Explicit_Type_Conversion.md

```diff
@@ -0,0 +1,92 @@
+# Explicit Type Conversion
+
+Type conversion is the process of converting data of one type to another.
+There are two types of type conversion in javascript:-
+Implicit conversion 
+Explicit conversion
+
+## Javascript Explicit Conversion
+
+- String to Numbers:
+
+The number () function is used to convert strings to numbers.
+parseInt() is used to return an integer by parsing strings.
+parseFloat() is used to returns a floating-point number.
+Empty strings to zero.
+Strings containing numbers to numbers
+Else converts to a NOT a NUMBER(NaN)
+
+The unary operator can be used to convert a variable to a number.
+
+- Number to Strings
+
+String() pr toString() function is used to convert numbers to strings for any expressions.
+toPrecision() is used to return a string of numbers having a specific length.
+
+- Dates to Numbers
+
+The number () function can be used to convert to numbers from date.
+
+- Booleans to Numbers/Strings
+
+The number () function is used for this conversion of booleans to numbers.
+For false it will return 0
+For true it will return 1
+
+Similarly, the String() function is used to convert booleans into strings.
+For false it will return false
+For true it will return true
+
+### Javascript Type Conversion Table
+
+| Value | Convert to string | Convert to number | Convert to boolean |
+|-----------|-------------------|-------------------|--------------------|
+| false | "false" | 0 | false |
+| true | "true" | 1 | true |
+| 0 | "0" | 0 | false |
+| 1 | "1" | 1 | true |
+| "1" | "1" | 1 | true |
+| "0" | "0" | 0 | true |
+| null | "null" | 0 | false |
+| undefined | "undefined" | NaN | false |
+| "ten" | "ten" | NaN | true |
+
+#### Javascript Implicit 
+
+In implicit form, it will automatically convert the value's type as needed.
+
+### Example illustrating the javascript Explicit type Conversion
+
+```
+<!DOCTYPE html>
+<html>
+<body>
+
+<h2>JavaScript Explicit Type Conversions</h2>
+
+<p>Converting null to other types:</p>
+
+<p id="demo" style="font-family:Times New Roman"></p>
+
+<script>
+let x = null;
+document.getElementById("demo").innerHTML =
+"Number : " + Number(x) + "<br>" +
+"String : " + String(x) + "<br>" +
+"Boolean: " + Boolean(x);
+</script>
+
+</body>
+</html>
+```
```

Comment:
code-blocks should have syntax for better readability

Reply:
@code-reaper08 Sir bold the subheadings but didn't get this review. Will you please explain this? So that I can resolve this as well!

Reply:
Hello @AS8420 👋 
By code-block syntax, what i mean is, for example,

1. This is what you've done, a normal code-block without syntax,

```
<!DOCTYPE html>
<html>
<body>
<h2>JavaScript Explicit Type Conversions</h2>
<p>Converting null to other types:</p>
<p id="demo" style="font-family:Times New Roman"></p>
<script>
let x = null;
document.getElementById("demo").innerHTML =
"Number : " + Number(x) + "<br>" +
"String : " + String(x) + "<br>" +
"Boolean: " + Boolean(x);
</script>
</body>
</html>
```

2. This is what is expected, code-blocks with syntax, this gives more readability.

```html
<!DOCTYPE html>
<html>
<body>
<h2>JavaScript Explicit Type Conversions</h2>
<p>Converting null to other types:</p>
<p id="demo" style="font-family:Times New Roman"></p>
<script>
let x = null;
document.getElementById("demo").innerHTML =
"Number : " + Number(x) + "<br>" +
"String : " + String(x) + "<br>" +
"Boolean: " + Boolean(x);
</script>
</body>
</html>
```

Reply:
@code-reaper08 Sir will I change those code with these one?